Automattic/wp-calypso
# packages: Remove `@automattic/react-i18n`

#### Changes proposed in this Pull Request

* Remove `react-i18n` package, as it's no longer being used anywhere since https://github.com/Automattic/wp-calypso/commit/e7a467cbfce95a2059c7b6aa770d8f9bc49b22bb

#### Testing instructions

* Confirm Calypso builds correctly.
* Check if all references to `@automattic/react-i18n` and its related files are removed.

Related to https://github.com/Automattic/wp-calypso/pull/51406
